### DISCUSSION TITLE
Six DOF Constraint question

### DISCUSSION THREAD

**Discussion by LeXXik** [jrouwe/JoltPhysics.js#162] [Q&A] [answered]
Hmm, I am trying to set up a six dof constraint.

I need to give an array of `SpringSettings` and `MotorSettings`:

https://github.com/jrouwe/JoltPhysics.js/blob/a36ddf07c891fa0896611c2421c1929ca656561e/JoltJS.idl#L1307-L1308

Not sure we can pass those.

**Comment by jrouwe** [comment]
You can just do e.g. `mLimitsSpringSettings[0].mFrequency = 20`?

**Reply by LeXXik** [reply]
Yep, this is what I do now. I grab the settings/frequency arrays and then access by indices. I think these 2 lines should be removed from IDL?
https://github.com/jrouwe/JoltPhysics.js/blob/a36ddf07c891fa0896611c2421c1929ca656561e/JoltJS.idl#L1307-L1308

**Reply by jrouwe** [reply]
I don't understand. If you remove these 2 lines then `mLimitsSpringSettings[0].mFrequency = 20` will no longer be possible because `mLimitsSpringSettings` will no longer exist.

**Reply by LeXXik** [reply]
Ah, I got confused for a moment with limits min and max. Nevermind me.

**Reply by LeXXik** [reply]
Hmm. Actually, I think I was accessing indices for something else. Those are not arrays, so I assume 1 value is used for all 6?

![image](https://github.com/jrouwe/JoltPhysics.js/assets/5677782/ad4f1011-fda9-4acb-80ad-60f86ccac6f9)

**Reply by LeXXik** [reply]
![image](https://github.com/jrouwe/JoltPhysics.js/assets/5677782/1b501805-8391-4f18-9c8e-a81c75fcbae6)

**Comment by LeXXik** [comment, marked as answer]
I think this is correct way. Can't use an accessor. Sorry for spam:

```js
settings.set_mMaxFriction(0, 1); //  set 1 on the first element
settings.get_mMaxFriction(0); // get the value of the first element

// similar for spring and motor settings:
settings.set_mLimitsSpringSettings(0, springSettings);
settings.set_mMotorSettings(0, motorSettings);
```

![image](https://github.com/jrouwe/JoltPhysics.js/assets/5677782/d029297f-53bd-4322-867d-0901423dddd1)
